checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java

```diff
@@ -846,7 +1121,7 @@ private void checkReassignmentToField(AssignmentNode node, Set<Obligation> oblig
     // extend beyond the method's body (and which therefore could not be targeted by an annotation
```

Comment:
A few lines above, at "static initializers don't need to be considered here": What about instance initializer blocks?

Reply:
The checker was unsound. I added a new test in this PR (`InstanceInitializers.java`) and some code here that defaults to issuing an error if an owning field with a non-empty must-call obligation is overwritten anywhere except:
* in a method, where we can use the existing procedure to check that the field has actually had its must-call obligations resolved first
* at the field's declaration (this is only sound-y in that it ignores the possibility that a superclass checks its own type and casts to a subclass in one of its constructors to assign the field, but that involves breaking the type system in other ways too so I think it's an acceptable trade-off)

Thanks for noticing this.